### PR TITLE
Make decodeAudioData more precise.

### LINKED ISSUE #1563: Make decodeAudioData more precise
It would be nice if https://webaudio.github.io/web-audio-api/#dom-baseaudiocontext-decodeaudiodata invoked an algorithm in MIME Sniffing rather than just referencing the whole standard. It would also be good to say which return values are accepted/rejected.

Are there tests for this method?

### PR DESCRIPTION
**Pull request by padenot**
This explicitely refers to the algorithm to use in mimesniff, and
adjusts the algorithm structure to be able to not attempt to decode if
the mimesniff algorithm failed.

This fixes #1563.

***
<a href="https://pr-preview.s3.amazonaws.com/padenot/web-audio-api/pull/1781.html" title="Last updated on Nov 6, 2018, 2:25 PM GMT (9ea0e6b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WebAudio/web-audio-api/1781/58b5ddd...padenot:9ea0e6b.html" title="Last updated on Nov 6, 2018, 2:25 PM GMT (9ea0e6b)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1132,12 +1132,21 @@ Methods</h4>
 			Note: Multiple <em>decoding threads</em> can run in parallel to
 			service multiple calls to <code>decodeAudioData</code>.
 
-			1. Attempt to decode the encoded {{BaseAudioContext/decodeAudioData(audioData, successCallback, errorCallback)/audioData!!argument}} into linear
-				PCM.
+			1. Let <var>canDecode</var> be a boolean flag, initially set to true
```
[the remark this answers is not in the document]
Comment (padenot) [reply index.bs]: `<var>` is semantically better. Does bikeshed static analysis work with `<em>` as well?

**Comment by padenot** [comment]
I updated this.

File: index.bs (hunk 1 above)
[the remark this answers is not in the document]
Comment (hoch) [reply index.bs]: SGTM. One more thing. Do we need to have a space between words like `can decode`? Some instances uses camel-casing and some don't. At some point, we would want to clean them up.

I do not know how Bikeshed behaves or should behave on `<em>`. I agree that `<var>` is much more meaningful here.

File: index.bs [hunk 2]
```diff
@@ -1132,12 +1132,21 @@ Methods</h4>
...
+			2. Attempt to determine the MIME type of
+				{{BaseAudioContext/decodeAudioData(audioData, successCallback,
+				errorCallback)/audioData!!argument}}, using
+				[[mimesniff#matching-an-audio-or-video-type-pattern]]. If the audio or
+				video type pattern matching algorithm returns <code>undefined</code>,
+				set <var>canDecode</var> to <em>false</em>.
+
+			3. If <var>canDecode</var> is <em>true</em>, attempt to decode the encoded
+				{{BaseAudioContext/decodeAudioData(audioData, successCallback,
+				errorCallback)/audioData!!argument}} into linear PCM. In case of
```
Comment (hoch) [inline index.bs]: Do we need a reference for "linear PCM" here?

File: index.bs [hunk 3]
```diff
@@ -1132,12 +1132,21 @@ Methods</h4>
 			Note: Multiple <em>decoding threads</em> can run in parallel to
 			service multiple calls to <code>decodeAudioData</code>.
 
-			1. Attempt to decode the encoded {{BaseAudioContext/decodeAudioData(audioData, successCallback, errorCallback)/audioData!!argument}} into linear
-				PCM.
+			1. Let <var>canDecode</var> be a boolean flag, initially set to true.
```
Comment (hoch) [inline index.bs]: Let's use `<var>can decode</var>`. Let us not camel-casing the algorithm variables from now on.

File: index.bs (hunk 2 above)
Comment (padenot) [reply index.bs]: Yes, but not in this PR.

File: index.bs (hunk 3 above)
Comment (padenot) [reply index.bs]: done.

**Pull request merged**